python_on_whales/components/container/cli_wrapper.py

```diff
@@ -563,31 +560,31 @@ def create(
         envs: Dict[str, str] = {},
         env_files: Union[ValidPath, List[ValidPath]] = [],
         expose: Union[int, List[int]] = [],
-        gpus: Union[int, str, None] = None,
+        gpus: Optional[Union[int, str]] = None,
         groups_add: List[str] = [],
         healthcheck: bool = True,
         health_cmd: Optional[str] = None,
-        health_interval: Union[None, int, timedelta] = None,
+        health_interval: Optional[Union[int, timedelta]] = None,
         health_retries: Optional[int] = None,
-        health_start_period: Union[None, int, timedelta] = None,
-        health_timeout: Union[None, int, timedelta] = None,
+        health_start_period: Optional[Union[int, timedelta]] = None,
+        health_timeout: Optional[Union[int, timedelta]] = None,
         hostname: Optional[str] = None,
         init: bool = False,
         ip: Optional[str] = None,
         ip6: Optional[str] = None,
         ipc: Optional[str] = None,
         isolation: Optional[str] = None,
-        kernel_memory: Union[int, str, None] = None,
+        kernel_memory: Optional[Union[int, str]] = None,
         labels: Dict[str, str] = {},
         label_files: List[ValidPath] = [],
         link: List[ValidContainer] = [],
         link_local_ip: List[str] = [],
         log_driver: Optional[str] = None,
         log_options: List[str] = [],
         mac_address: Optional[str] = None,
-        memory: Union[int, str, None] = None,
-        memory_reservation: Union[int, str, None] = None,
-        memory_swap: Union[int, str, None] = None,
+        memory: Optional[Union[int, str]] = None,
```

Comment:
I'm not going to block this pull request for this, but note that this is not very useful to change, next python's release, we'll have `int | str | None` instead.

Reply:
Yeah sorry this was kind of a style preference change, I personally prefer using `Optional` to indicate arguments where `None` is used to represent "not specified" as opposed to having some other meaning (such as "no timeout"). It's then consistent with the other args that are `Optional[<something>]`, and doesn't look like the `| None` was missed.

I can change these back if you don't like this.

Reply:
No worries, I won't block the PR for this, I'm just saying that it's not useful as it's very temporary.

Reply:
Part of what I was saying is that even with the new syntax available I prefer sticking to `Optional` here, due to how it reads. That is, I'm agreeing with [this perspective](https://discuss.python.org/t/clarification-for-pep-604-is-foo-int-none-to-replace-all-use-of-foo-optional-int/26945/10).

Reply:
We can't let someone opinion be a standard for an implied meaning, notably because not enough people know about it so it's not going to be applied throughout the codebase, so the meaning in the codebase of `Optional[x]` won't be reliable. Even if the convention comes from a CPython core dev or other. 

As long as it's not in the python documentation or not in a PEP or not in a linter tool to enforce it, we can only assume true what is written in the python docs, that being that `Optional[x] == x | None`. Since the second option is shorter and doesn't require an import, it's going to be preferred in all cases thoughout the project, especially since we can enforce it automatically. We'll likely enforce it with pyupgrade or something similar because memorizing conventions when doing code reviews is a pain.

Long story short, if we adopt the convention proposed here, we have no way of enforce it thoughout the project, unless we force reviewers to force it, and it's very annoying. More automation is always preferable, we'll use pyupgrade for this.